### ISSUE TITLE
Compiler error with Java 21

### ISSUE THREAD

**Issue by cushon** [tony19/named-regexp#73]
### Describe the bug

The project fails to compile at the Java 21 language level.

### Reproduction

JAVA_HOME=<path to JDK 21> mvn install

### Logs

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (default-compile) on project named-regexp: Compilation failure: Compilation failure:
[ERROR] ./named-regexp/src/main/java/com/google/code/regexp/Matcher.java:[314,38] namedGroups() in com.google.code.regexp.Matcher cannot implement namedGroups() in java.util.regex.MatchResult
[ERROR]   return type java.util.List<java.util.Map<java.lang.String,java.lang.String>> is not compatible with java.util.Map<java.lang.String,java.lang.Integer>
[ERROR] .,/named-regexp/src/main/java/com/google/code/regexp/MatchResult.java:[44,38] namedGroups() in com.google.code.regexp.MatchResult clashes with namedGroups() in java.util.regex.MatchResult
[ERROR]   return type java.util.List<java.util.Map<java.lang.String,java.lang.String>> is not compatible with java.util.Map<java.lang.String,java.lang.Integer>
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by cushon** [comment]
The `namedGroups()` API was added to the JDK in JDK 20: https://bugs.openjdk.org/browse/JDK-8292872

The repro also required updating the source version to 1.8:

```diff
diff --git a/pom.xml b/pom.xml
index c88cba2..2f96e9d 100644
--- a/pom.xml
+++ b/pom.xml
@@ -60,8 +60,8 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.javadoc.failOnError>false</maven.javadoc.failOnError>
-    <maven.compiler.source>1.6</maven.compiler.source>
-    <maven.compiler.target>1.6</maven.compiler.target>
+    <maven.compiler.source>8</maven.compiler.source>
+    <maven.compiler.target>8</maven.compiler.target>
   </properties>

   <build>
```

**Comment by cushon** [comment]
Thanks, that is a good question! I noticed this because my employer builds third party dependencies from source, and I was testing with Java 21. We were using named-regexp as a dependency of https://github.com/contentful/contentful.java

Using `java.util.regex` instead of the backport works for me, I also sent a PR to update contentful: https://github.com/contentful/contentful.java

**Comment by tony19** [comment]
This is fixed in #74, released in [v1.0.0](https://github.com/tony19/named-regexp/releases/tag/named-regexp-1.0.0).
